traffic_portal/app/src/common/api/UserService.js

```diff
@@ -17,143 +17,152 @@
-                    locationUtils.navigateToPath('/users');
-                },
-                function(fault) {
-                    messageModel.setMessages(fault.data.alerts, false);
-                }
-            );
+        return $http.post(ENV.api['root'] + 'users', user).then(
+            function(result) {
+                messageModel.setMessages([ { level: 'success', text: 'User created' } ], true);
+                locationUtils.navigateToPath('/users');
+                return result;
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, false);
+                throw err;
+            }
+        );
     };
 
     this.updateUser = function(user) {
-        return $http.put(ENV.api['root'] + "users/" + user.id, user)
-            .then(
-                function(result) {
-                    if (userModel.user.id == user.id) {
-                        // if you are updating the currently logged in user...
-                        userModel.setUser(user);
-                    }
-                    messageModel.setMessages(result.data.alerts, false);
-                },
-                function(fault) {
-                    messageModel.setMessages(fault.data.alerts, false);
+        return $http.put(ENV.api['root'] + "users/" + user.id, user).then(
+            function(result) {
+                if (userModel.user.id === user.id) {
+                    // if you are updating the currently logged in user...
+                    userModel.setUser(user);
                 }
-            );
+                messageModel.setMessages(result.data.alerts, false);
+                return result;
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, false);
+                throw err;
+            }
+        );
     };
 
     this.updateCurrentUser = function(user) {
-        return $http.post(ENV.api['root'] + "user/current/update", { user: user })
-            .then(
-                function() {
-                    userModel.setUser(user);
-                    messageModel.setMessages([ { level: 'success', text: 'Current user updated' } ], false);
-                },
-                function() {
-                    messageModel.setMessages([ { level: 'error', text: 'Current user updated failed' } ], false);
-                }
-            );
+        return $http.post(ENV.api['root'] + "user/current/update", { user: user }).then(
+            function(result) {
+                userModel.setUser(user);
+                messageModel.setMessages([ { level: 'success', text: 'Current user updated' } ], false);
+                return result;
+            },
+            function(err) {
+                messageModel.setMessages([ { level: 'error', text: 'Current user updated failed' } ], false);
+                throw err;
+            }
+        );
     };
 
     this.getUnassignedUserDeliveryServices = function(userId) {
```

Comment:
can you delete these methods as they are not used anywhere:

getUnassignedUserDeliveryServices
deleteUserDeliveryService
assignUserDeliveryServices

Reply:
done